lib/game_modules/player.ex

```diff
@@ -1,4 +1,14 @@
 defmodule Player do
 
+  def create_player(icon, type) do
+    %{icon: icon, type: type}
```

Comment:
Maps are a good start here. When I see a `type` field, it's a little bit of a code smell: it usually ends with a ["replace conditional with polymorphism"](https://refactoring.guru/replace-conditional-with-polymorphism) refactor.

Different types of players will behave differently but conform to the same interface. That's a great case for polymorphism. A simple map with a `type` attribute is a fine place to start, but hopefully you can lean on some of the tools Elixir provides to implement polymorphic players.

I'd start by reading up on [structs](https://elixir-lang.org/getting-started/structs.html), [protocols](https://elixir-lang.org/getting-started/protocols.html), and [polymorphism in Elixir](https://medium.com/elixirlabs/polymorphism-in-elixir-cd0c765b6929). Check out chapter 24 in the Elixir book too.